File: GXpert/GXpert.Web/Modules/Activation/SerialKey/SerialKeyColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Activation.Columns;

[ColumnsScript("Activation.SerialKey")]
[BasedOnRow(typeof(SerialKeyRow), CheckNames = true)]
public class SerialKeyColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string SerialKey { get; set; }
    public string PlayListTitle { get; set; }
    [QuickFilter]
    public short ValidityType { get; set; }
    public short ValidityInDays { get; set; }
    public DateTime ValidDate { get; set; }
    public string Note { get; set; }
    [QuickFilter]
    public short EStatus { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Activation/SerialKeyColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EValidityType } from "../Web/Enums.EValidityType";
import { SerialKeyRow } from "./SerialKeyRow";

export interface SerialKeyColumns {
    Id: Column<SerialKeyRow>;
    SerialKey: Column<SerialKeyRow>;
    PlayListTitle: Column<SerialKeyRow>;
    ValidityType: Column<SerialKeyRow>;
    ValidityInDays: Column<SerialKeyRow>;
    ValidDate: Column<SerialKeyRow>;
    Note: Column<SerialKeyRow>;
    EStatus: Column<SerialKeyRow>;
}

export class SerialKeyColumns extends ColumnsBase<SerialKeyRow> {
    static readonly columnsKey = 'Activation.SerialKey';
    static readonly Fields = fieldsProxy<SerialKeyColumns>();
}

[EValidityType]; // referenced types